File: Front.QuizApp/src/app/business/mappers/questionIndexAndId.mapper.ts
import {Injectable} from '@angular/core';
import {QuestionIndexAndIdDto} from '../dtos/questionIndexAndId.dto';
import {QuestionIndexAndId} from '../models/questionIndexAndId.model';

@Injectable({providedIn: 'root'})
export class QuestionIndexAndIdMapper {
    public mapFromApiToModel(apiQuestionIndexAndId: QuestionIndexAndIdDto): QuestionIndexAndId {
        return {
            questionId: apiQuestionIndexAndId.questionId,
            totalNumberOfQuestions: apiQuestionIndexAndId.totalNumberOfQuestions,
            currentQuestionIndex: apiQuestionIndexAndId.currentQuestionIndex,
        };
    }

}

File: Back.QuizApp/Controllers/Mappers/QuestionIndexAndIdMapper.cs
using Ynov.QuizApp.Business.Models;
using Ynov.QuizApp.Controllers.DTOs;

namespace Ynov.QuizApp.Controllers.Mappers;

public class QuestionIndexAndIdMapper {
    public QuestionIndexAndIdDTO ToDto(QuestionIndexAndId questionIndexAndId) {
        return new QuestionIndexAndIdDTO() {
            QuestionId = questionIndexAndId.QuestionId,
            TotalNumberOfQuestions = questionIndexAndId.TotalNumberOfQuestions,
            CurrentQuestionIndex = questionIndexAndId.CurrentQuestionIndex
        }; 
    }
}